lib/sort_by_length.rb

```diff
@@ -1,7 +1,20 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) - loops dependent on length of input
+# Space complexity: O(n) - size of array is dependent on input
 def sort_by_length(my_sentence)
```

Comment:
Take a look at the implementations for sorting I gave as an example, as this doesn't actually sort things in general.  You can't sort a generic array with only one pass through the array.  It looks like you started bubblesort, but didn't put the outer loop in.